mathesar_ui/src/systems/data-explorer/utils.ts

```diff
@@ -465,52 +457,46 @@ export function speculateColumnMetaData({
-                type_options: inputColumn?.type_options ?? null,
-                metadata: inputColumn?.metadata ?? null,
-                is_initial_column: false,
-                input_alias: group.inputAlias,
-              },
-              abstractTypeMap,
-            ),
+            processColumn({
+              alias: group.outputAlias,
+              display_name: null,
+              type: inputColumn?.type ?? 'unknown',
+              type_options: inputColumn?.type_options ?? null,
+              metadata: inputColumn?.metadata ?? null,
+              is_initial_column: false,
+              input_alias: group.inputAlias,
+            }),
           );
         }
       });
       [...transform.aggregations.values()].forEach((aggregation) => {
         if (!updatedColumnsMetaData.has(aggregation.outputAlias)) {
           updatedColumnsMetaData = updatedColumnsMetaData.with(
             aggregation.outputAlias,
-            processColumn(
-              {
-                alias: aggregation.outputAlias,
-                type:
-                  aggregation.function === 'distinct_aggregate_to_array'
-                    ? '_array'
-                    : 'integer',
-                type_options:
-                  aggregation.function === 'distinct_aggregate_to_array'
-                    ? {
-                        // TODO_BETA: Ask Pavish.
-                        //
-                        // `Column['type_options']` was previously typed loosely
-                        // as `Record<string, unknown> | null`. Now it's more
-                        // strict and it doesn't have a `type` property.
-                        //
-                        // type:
-                        //   updatedColumnsMetaData.get(aggregation.inputAlias)
-                        //     ?.column.type ?? 'unknown',
-                      }
-                    : null,
-                metadata: null,
-                is_initial_column: false,
-                input_alias: aggregation.inputAlias,
-              },
-              abstractTypeMap,
-            ),
+            processColumn({
+              alias: aggregation.outputAlias,
+              type:
+                aggregation.function === 'distinct_aggregate_to_array'
+                  ? '_array'
+                  : 'integer',
+              type_options:
+                aggregation.function === 'distinct_aggregate_to_array'
+                  ? {
+                      // TODO_BETA: Ask Pavish.
+                      //
+                      // `Column['type_options']` was previously typed loosely
+                      // as `Record<string, unknown> | null`. Now it's more
+                      // strict and it doesn't have a `type` property.
+                      //
+                      // type:
+                      //   updatedColumnsMetaData.get(aggregation.inputAlias)
+                      //     ?.column.type ?? 'unknown',
+                    }
```

Comment:
We discussed this on a 1:1 before. I think we missed making the code change.

`type` here is now changed to `item_type`. This applies for `_array` types which require the type of items they hold.